### ISSUE TITLE
Future parser loop should allow for inline variable declaration without scoping

### ISSUE THREAD

**Issue by favoretti** [rodjek/puppet-lint#264]
Consider the following code:

```
    $nsconfig['csvservers'].each |$csvserver| {
      nagios_checks::hardware::netscaler {
        "csvserver_${csvserver}":
          lbip      => $nsfqdn,
          user      => $nsconfig['user'],
          password  => $nsconfig['pass'],
          csvserver => $csvserver;
      }
    }
```

This produces the following warning:

```
WARNING: top-scope variable being used without an explicit namespace on line 63
$nsconfig['csvservers'].each |$csvserver| {
                                               ^
```

Technically `$csvserver` is a loop-local variable, hence should not be scoped (IMHO, correct me if I'm wrong). Working this around is done by explicitly declaring variable before loop:

```
$csvserver = ''
```

I think piece of code above should not generate a warning and not force to declare useless variables. Opinions?
